src/AppInstallerRepositoryCore/Microsoft/Schema/Checkpoint_1_0/CheckpointTable.cpp

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "CheckpointTable.h"
+#include "SQLiteStatementBuilder.h"
+
+namespace AppInstaller::Repository::Microsoft::Schema::Checkpoint_V1_0
+{
+    using namespace std::string_view_literals;
+    static constexpr std::string_view s_CheckpointTable_Table_Name = "Checkpoints"sv;
+    static constexpr std::string_view s_CheckpointTable_Name_Column = "Name";
+    static constexpr std::string_view s_CheckpointTable_WriteTime_Column = "WriteTime";
+
+    std::string_view CheckpointTable::TableName()
+    {
+        return s_CheckpointTable_Table_Name;
+    }
+    
+    void CheckpointTable::Create(SQLite::Connection& connection)
+    {
+        using namespace SQLite::Builder;
+
+        SQLite::Savepoint savepoint = SQLite::Savepoint::Create(connection, "createCheckpointTable_v1_0");
+
+        StatementBuilder createTableBuilder;
+        createTableBuilder.CreateTable(s_CheckpointTable_Table_Name).BeginColumns();
+        createTableBuilder.Column(ColumnBuilder(s_CheckpointTable_Name_Column, Type::Text).Unique());
```

Comment:
Any time that you are going to use the row id as a foreign key in another table, you must specify it explicitly as a column or it can be changed on you when the table is rebalanced.  See the index code for the way that you do it (it has to be one of a few specific names and with specific properties).

Reply:
Removed Guid and just made the resume id a string. Also removed the argument validation so that it simply checks if we can find the checkpoint database with the provided resume id string.